engine/src/main/java/org/terasology/engine/logic/behavior/BehaviorComponent.java

```diff
@@ -2,16 +2,22 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.engine.logic.behavior;
 
-import org.terasology.engine.entitySystem.Component;
 import org.terasology.engine.logic.behavior.asset.BehaviorTree;
+import org.terasology.gestalt.entitysystem.component.Component;
 import org.terasology.gestalt.module.sandbox.API;
 
 /**
  * Entities with this component are handled by a behavior tree. Default tree to fetch may be set.
  *
  */
 @API
-public class BehaviorComponent implements Component {
+public class BehaviorComponent implements Component<BehaviorComponent> {
     public BehaviorTree tree;
     public transient Interpreter interpreter;
+
+    @Override
+    public void copy(BehaviorComponent other) {
+        this.tree = other.tree;
+        this.interpreter = other.interpreter;
```

Comment:
`Interpreter` has a copy constructor to create a new instance, but does not do deep copies of any of its members ( e.g., actor or tree). Since it has a copy constructor, should we use it for good measure?

^ @casals any input on this?